lib/kitchen/util.rb

```diff
@@ -141,5 +141,59 @@ def self.shell_helpers
                 File.dirname(__FILE__), %w{.. .. support download_helpers.sh}
       ))
     end
+
+    # Lists the contents of the given directory. path will be prepended
+    # to the list returned. '.' and '..' are never returned.
+    #
+    # @param path [String] the directory to list
+    # @param include_dot [Boolean] if true, dot files will be included
+    # @param recurse [Boolean] if true, listing will be recursive
+    # @return A listing of the specified path
+    #
+    # @note You should prefer this method to using Dir.glob directly. The reason is
+    # because Dir.glob behaves strangely on Windows. It wont accept '\'
+    # and doesn't like fake directories (C:\Documents and Settings)
+    # It also does not do any sort of error checking, so things one would
+    # expect to fail just return an empty list
+    def self.list_directory(path, include_dot: false, recurse: false)
+      # Things (such as tests) are relying on this to not blow up if
+      # the directory does not exist
+      return [] if !Dir.exist?(path)
+
+      Dir.chdir(path) do
+        glob_pattern = if recurse
+                         "**/*"
+                       else
+                         "*"
+                       end
+        flags = if include_dot
+                  [File::FNM_DOTMATCH]
+                else
+                  []
+                end
+        Dir.glob(glob_pattern, *flags)
```

Comment:
Would it make sense to use `safe_glob` here instead of re-doing `Dir.chdir { Dir.glob }`? Or would that make it harder to filter out `.` and `..`?